Automattic/wp-calypso
# Import: Skip plans page in JPC flow if user comes from migration flow

#### Proposed Changes

We don't need to upsell Jetpack plans if a user is coming to JPC flow from the migration flow, the purpose of this PR is to skip the plans page if a user is from migration flow. To do it, we'll need to:
* Adding a `source` flag into JPC flow so we can know if it's from the migration flow.
* Close the JPC flow window after a site is fully connected if a user is coming from the migration flow.
* Checking if a user owns the site when the user arrives at the beginning of JPC flow and see if they're from migration flow. If so, we will close the window.

Before:
**Test case 1: Fully disconnected site(this means your self-hosted site is not connected to WP.com and has no account connection)**:

-  It redirects you to the plans page after you're fully connected.

https://user-images.githubusercontent.com/4074459/187843994-5da6a8a2-bea3-4361-8bec-02ae743dd944.mp4

**Test case 2: Half connected site(this means your self-hosted site is connected to WP.com but no account connection)**:
-  It redirects you to the plans page after you're in JPC flow.

https://user-images.githubusercontent.com/4074459/187844287-c36a98a0-e0fa-4d68-8ca3-ea6641c36f9c.mov

#### Testing Instructions

**Test case 1: Fully disconnected site(this means your self-hosted site is not connected to WP.com and has no account connection)**:
* Expected behavior: It should close the JPC flow window after your site is fully connected. 
1. Creating a self-hosted site via [jurassic-ninja](https://jurassic.ninja/) or use your own self-hosted site(should be disconnected).
2. Navigate to `http://calypso.localhost:3000/setup/import?siteSlug=${SITE_SLUG}`.
3. Place the url of the test self-hosted site and click Continue -> Import your content -> Install Jetpack.
4. It should open a new tab for JPC flow, the url should be something like `http://calypso.localhost:3000/jetpack/connect?url=${YOUR_SELF_HOSTED_SITE}&source=import` and it'll run the site connection automatically.
5. If you haven't login to your self-hosted site before, it'll land on a login screen as asking the credential of your self-hosted site. Please don't enter the credential yet. Beware that at this point your url is `https://wordpress.com/jetpack/install`. For testing purposes, we need to replace the url with `http://calypso.localhost:3000/jetpack/connect?forceInstall=1&url=${YOUR_SELF_HOSTED_SITEURL}` 
6. Enter your login credential for your self-hosted site and click the Install Jetpack button.
7. If you're facing 503 on your test Jurassic Ninja site please ignore it and refresh the page again.
8. You'll be landed on a completing setup page. Please replace the `https://wordpress.com` part with `http://calypso.localhost:3000` then hit the Approve button. 
9. Once it's finished the JPC flow window will close by itself and you can continue with the importing process.

If you'd like to see the testing process in action, please check out the video below:

https://user-images.githubusercontent.com/4074459/188078133-50a65e8f-8238-420a-aedd-475aa4ba3b68.mp4

**Test case 2: Half connected site(this means your self-hosted site is connected to WP.com but no account connection)**:
* Expected behavior: It should redirect you to the authorization page after your are in JPC flow.
1. To prepare a half-connected site, you can navigate to `http://calypso.localhost:3000/jetpack/connect` and enter your self-hosted site and click Continue. 
2. Once you are on `/jetpack/connect/authorize` page don't click the approve button and close the window. This leaves your site half-connected.
3. Go through step 2 to 4 again, this time you'll be redirected to the authorize page instead of the plans page. The rest steps are same as Test case 1.

If you'd like to see the testing process in action, please check out the video below:

https://user-images.githubusercontent.com/4074459/187842698-5dac34af-8936-4484-b08c-4f406ff6499b.mov

#### Pre-merge Checklist




Related to #67023
